diff --git a/src/decompiler/printc.cc b/src/decompiler/printc.cc
index b2637c1..eac56f7 100644
--- a/src/decompiler/printc.cc
+++ b/src/decompiler/printc.cc
@@ -54,7 +54,7 @@ OpToken PrintC::boolean_and = { "&&", "", 2, 22, false, OpToken::binary, 1, 0, (
 OpToken PrintC::boolean_xor = { "^^", "", 2, 20, false, OpToken::binary, 1, 0, (OpToken *)0 };
 OpToken PrintC::boolean_or = { "||", "", 2, 18, false, OpToken::binary, 1, 0, (OpToken *)0 };
 OpToken PrintC::assignment = { "=", "", 2, 14, false, OpToken::binary, 1, 5, (OpToken *)0 };
-OpToken PrintC::comma = { ",", "", 2, 2, true, OpToken::binary, 0, 0, (OpToken *)0 };
+OpToken PrintC::comma = { ", ", "", 2, 2, true, OpToken::binary, 0, 0, (OpToken *)0 };
 OpToken PrintC::new_op = { "", "", 2, 62, false, OpToken::space, 1, 0, (OpToken *)0 };
 
 // Inplace assignment operators
@@ -82,7 +82,7 @@ const string PrintC::CLOSE_CURLY = "}";
 const string PrintC::SEMICOLON = ";";
 const string PrintC::COLON = ":";
 const string PrintC::EQUALSIGN = "=";
-const string PrintC::COMMA = ",";
+const string PrintC::COMMA = ", ";
 const string PrintC::DOTDOTDOT = "...";
 const string PrintC::KEYWORD_VOID = "void";
 const string PrintC::KEYWORD_TRUE = "true";