diff --git a/src/decompiler/xml.cc b/src/decompiler/xml.cc
index 6831da5..fb4a87c 100644
--- a/src/decompiler/xml.cc
+++ b/src/decompiler/xml.cc
@@ -101,7 +101,6 @@
 
 namespace ghidra {
 
-string Attributes::bogus_uri("http://unused.uri");
 
 /// \brief The XML character scanner
 ///
diff --git a/src/decompiler/xml.hh b/src/decompiler/xml.hh
index 7253301..3fff761 100644
--- a/src/decompiler/xml.hh
+++ b/src/decompiler/xml.hh
@@ -42,8 +42,8 @@ using std::hex;
 /// This object is used to initialize the Element object but is not part of the final, in memory, DOM model.
 /// This also holds other properties of the element that are unused in this implementation,
 /// including the \e namespace URI.
+#define bogus_uri "http://unused.uri"
 class Attributes {
-  static string bogus_uri;		///< A placeholder for the namespace URI that should be attached to the element
 //  static string prefix;
   string *elementname;			///< The name of the XML element
   vector<string *> name;		///< List of names for each formal XML attribute
