diff --git a/src/decompiler/database.cc b/src/decompiler/database.cc
index 6eab457..8d17f38 100644
--- a/src/decompiler/database.cc
+++ b/src/decompiler/database.cc
@@ -1805,9 +1805,10 @@ SymbolEntry *ScopeInternal::addMapInternal(Symbol *sym,uint4 exfl,const Address
     string msg = "Symbol ";
     msg += sym->getName();
     msg += " extends beyond the end of the address space";
-    throw LowlevelError(msg);
+    fprintf (stderr, "%s\n", msg.c_str());
+    //throw LowlevelError(msg);
   }
-    
+
   list<SymbolEntry>::iterator iter = rangemap->insert(initdata,addr.getOffset(),lastaddress.getOffset());
   // Store reference to map in symbol
   sym->mapentry.push_back(iter);
