diff --git a/src/decompiler/prettyprint.cc b/src/decompiler/prettyprint.cc
index f85f018..1ec21f3 100644
--- a/src/decompiler/prettyprint.cc
+++ b/src/decompiler/prettyprint.cc
@@ -345,7 +345,7 @@ void EmitMarkup::setOutputStream(ostream *t)
   if (encoder != (Encoder *)0)
     delete encoder;
   s = t;
-  encoder = new PackedEncode(*s);
+  encoder = new XmlEncode(*s);
 }
 
 void EmitMarkup::setPackedOutput(bool val)
