diff --git a/src/decompiler/database.cc b/src/decompiler/database.cc
index 26b68a6..0089445 100644
--- a/src/decompiler/database.cc
+++ b/src/decompiler/database.cc
@@ -140,6 +140,8 @@ Datatype *MapEntry::getSizedType(const Address &inaddr,int4 sz) const
 {
   int4 off = (int4)(inaddr.getOffset() - addr.getOffset()) + offset;
   Datatype *cur = symbol->getType();
+  if (symbol->getName().length() == 0)
+    return nullptr;
   return symbol->getScope()->getArch()->types->getExactPiece(cur, off, sz);
 }
 
diff --git a/src/decompiler/varmap.cc b/src/decompiler/varmap.cc
index 6d51a72..0388f3b 100644
--- a/src/decompiler/varmap.cc
+++ b/src/decompiler/varmap.cc
@@ -1375,7 +1375,7 @@ void ScopeLocal::markUnaliased(const vector<uintb> &alias)
     // NOTE: this is primarily to reset aliasing between
     // stack parameters and stack locals
     if (aliason && (curoff - curalias > 0xffff)) aliason = false;
-    if (!aliason) symbol->getScope()->setAttribute(symbol,Varnode::nolocalalias);
+    if (!aliason && symbol->getName().length () > 0) symbol->getScope()->setAttribute(symbol,Varnode::nolocalalias);
     if (symbol->isTypeLocked() && alias_block_level != 0) {
       if (alias_block_level == 3)
 	aliason = false;		// For this level, all locked data-types block aliases
