diff --git a/src/decompiler/slghparse.hh b/src/decompiler/slghparse.hh
index 5d72496..d798b4b 100644
--- a/src/decompiler/slghparse.hh
+++ b/src/decompiler/slghparse.hh
@@ -166,7 +166,7 @@ extern int sleighdebug;
     ELLIPSIS_KEY = 344,            /* ELLIPSIS_KEY  */
     GLOBALSET_KEY = 345,           /* GLOBALSET_KEY  */
     BITRANGE_KEY = 346,            /* BITRANGE_KEY  */
-    CHAR = 347,                    /* CHAR  */
+    CHARVALUE = 347,               /* CHAR  */
     INTEGER = 348,                 /* INTEGER  */
     INTB = 349,                    /* INTB  */
     STRING = 350,                  /* STRING  */
diff --git a/src/decompiler/slghscan.cc b/src/decompiler/slghscan.cc
index a8545a5..9709240 100644
--- a/src/decompiler/slghscan.cc
+++ b/src/decompiler/slghscan.cc
@@ -2129,7 +2129,7 @@ YY_RULE_SETUP
 	YY_BREAK
 case 61:
 YY_RULE_SETUP
-{ sleighlval.ch = sleightext[0]; return CHAR; }
+{ sleighlval.ch = sleightext[0]; return CHARVALUE; }
 	YY_BREAK
 case 62:
 YY_RULE_SETUP
